src/GraphTopology.cpp

```diff
@@ -424,28 +349,12 @@ void GraphTopology::Init(const mfem::Array<int>& partition,
     SetConstantValue(*face_trueface_face_, 1.0);
 
     // Construct "face to true face" table
-    face_trueface_ = BuildEntityToTrueEntity(*face_trueface_face_);
-}
-
-std::vector<GraphTopology> MultilevelGraphTopology(
-    const Graph& graph, const mfem::SparseMatrix* edge_boundaryattr,
-    int num_levels, int coarsening_factor)
-{
-    std::vector<GraphTopology> graph_topologies;
-    graph_topologies.reserve(num_levels - 1);
-
-    // Construct finest level graph topology
-    mfem::Array<int> partitioning;
-    PartitionAAT(graph.GetVertexToEdge(), partitioning, coarsening_factor);
-    graph_topologies.emplace_back(graph, partitioning, edge_boundaryattr);
-
-    // Construct coarser levels graph topology by recursion
-    for (int i = 0; i < num_levels - 2; i++)
-    {
-        graph_topologies.emplace_back(graph_topologies.back(), coarsening_factor);
-    }
+    auto face_trueface = BuildEntityToTrueEntity(*face_trueface_face_);
 
-    return graph_topologies;
+    auto coarse_graph = make_unique<Graph>(
+                            Agg_face, *face_trueface, mfem::Vector(), face_bdratt.get());
+    coarse_graph_ = coarse_graph.get();
```

Comment:
I don't like this. You make a `unique_ptr`, then get a raw pointer, and you pretend to return a `unique_ptr` but it's not really unique because this class retains a (publicly accessible!) raw pointer to the same object.